app/models/host/managed.rb

```diff
@@ -961,7 +962,7 @@ def assign_hostgroup_attributes(attrs = [])
     attrs.each do |attr|
       next if send(attr).to_i == -1
       value = hostgroup.send("inherited_#{attr}")
-      self.send("#{attr}=", value) unless send(attr).present?
+      self.send("#{attr}=", value)
```

Comment:
I think the test is failing because of this line lacking `unless send(attr).present?` - this preserves the behavior we had, override only empty values.